In [ ]:
!pip install catboost xgboost scikit-learn --quiet

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 99.2/99.2 MB 10.1 MB/s eta 0:00:00


In [ ]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
from sklearn.ensemble import RandomForestRegressor, StackingRegressor
from sklearn.linear_model import Ridge
from xgboost import XGBRegressor
from catboost import CatBoostRegressor
import matplotlib.pyplot as plt

In [ ]:
import joblib

# 모델 경로
model_path = '/content/drive/MyDrive/DS teamproj/stacking_model_01.pkl'

# 모델 로드
loaded_model = joblib.load(model_path)
print("✅ 모델이 불러와졌습니다.")


✅ 모델이 불러와졌습니다.


In [ ]:
result=[]
expected_columns = loaded_model.feature_names_in_

for i in range(1,8):
  pred_y = []
  df_test = pd.read_csv(f'/content/drive/MyDrive/DS teamproj/cleaned_test_h{i}_202403290800.csv')

  # 누락된 season 관련 더미 컬럼 생성 (모두 False로 초기화)
  for col in ['season_summer', 'season_fall', 'season_winter']:
      if col not in df_test.columns:
          df_test[col] = False
  # 순서 맞추기
  df_test = df_test[expected_columns]

  y_pred = loaded_model.predict(df_test)
  mean_y = float(y_pred.mean())
  result.append(mean_y)


In [ ]:
print(result)

[404.8529052734375, 409.7353820800781, 410.4626159667969, 410.8769836425781, 407.6650695800781, 412.81121826171875, 408.3542175292969]


In [ ]:
decreasing = []
for a in result :
  a = 439.56 - a
  decreasing.append(a)

print(decreasing)
max_value = max(decreasing)
max_index = decreasing.index(max_value)+1
print(f"가장 큰 평균 감소값: {max_value}, 위치: h{max_index}")

[34.7070947265625, 29.824617919921877, 29.097384033203127, 28.683016357421877, 31.894930419921877, 26.748781738281252, 31.205782470703127]
가장 큰 평균 감소값: 34.7070947265625, 위치: h1


기존 평균 pm10 = 439.56

**예측값**
[404.8529052734375, 409.7353820800781, 410.4626159667969, 410.8769836425781, 407.6650695800781, 412.81121826171875, 408.3542175292969]


**감소량**
[34.7070947265625, 29.824617919921877, 29.097384033203127, 28.683016357421877, 31.894930419921877, 26.748781738281252, 31.205782470703127]

가장 큰 평균 감소값: 34.7070947265625,  
##위치: h1